src/rwe/cob/CobEnvironment.cpp

```diff
@@ -35,15 +35,15 @@ namespace rwe
         return createNonScheduledThread(index, params);
     }
 
-    CobThread CobEnvironment::createNonScheduledThread(unsigned int functionId, const std::vector<int>& params)
+    CobThread CobEnvironment::createNonScheduledThread(int functionId, const std::vector<int>& params)
     {
         const auto& functionInfo = _script->functions.at(functionId);
         CobThread thread(functionInfo.name);
         thread.callStack.emplace(functionInfo.address, params);
         return thread;
     }
 
-    const CobThread* CobEnvironment::createThread(unsigned int functionId, const std::vector<int>& params, unsigned int signalMask)
+    const CobThread* CobEnvironment::createThread(int functionId, const std::vector<int>& params, int signalMask)
```

Comment:
I think here it is more appropriate for `signalMask` (and `signal`) to remain unsigned. The C++ Core Guidelines ES.101 recommends unsigned when the value will be used for bit manipulation, which `signal` and `signalMask` are, as they are ANDed together.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-unsigned